libbeat/monitoring/report/elasticsearch/client.go

```diff
@@ -26,30 +26,34 @@ import (
 	"github.com/pkg/errors"
 
 	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/esleg/eslegclient"
 	"github.com/elastic/beats/v7/libbeat/logp"
 	"github.com/elastic/beats/v7/libbeat/monitoring/report"
-	esout "github.com/elastic/beats/v7/libbeat/outputs/elasticsearch"
 	"github.com/elastic/beats/v7/libbeat/publisher"
 	"github.com/elastic/beats/v7/libbeat/testing"
 )
 
 var createDocPrivAvailableESVersion = common.MustNewVersion("7.5.0")
 
 type publishClient struct {
-	es     *esout.Client
+	es     *eslegclient.Connection
 	params map[string]string
 	format report.Format
+
+	log *logp.Logger
 }
 
 func newPublishClient(
-	es *esout.Client,
+	es *eslegclient.Connection,
 	params map[string]string,
 	format report.Format,
 ) (*publishClient, error) {
 	p := &publishClient{
 		es:     es,
```

Comment:
There will be an extra line removal here once https://github.com/elastic/beats/pull/17211 is merged (and this PR here is rebased on `7.x`).